webapp/Connector/src/utility/PlotTooltip.js

```diff
@@ -0,0 +1,471 @@
+                            values = dimensions[dim];
+                        }
+                    }
+                    if (values.length > 0) {
+                        if (hierarchicalDimensionInfo[dim]) {
+                            var levelDimensions = hierarchicalDimensionInfo[dim];
+                            var dimensionVals = {};
+                            Ext.each(values, function(fullStr){
+                                var levels = fullStr.split(ChartUtils.ANTIGEN_LEVEL_DELIMITER);
+                                for (var i = 0; i < levels.length; i++) {
+                                    if (!dimensionVals[levelDimensions[i]]) {
+                                        dimensionVals[levelDimensions[i]] = {};
+                                    }
+                                    dimensionVals[levelDimensions[i]][levels[i]] = true;
+                                }
+                            });
+
+                            for (var level in dimensionVals) {
+                                if (dimensionVals[level]) {
+                                    var levelValues = Object.keys(dimensionVals[level]);
+                                    if (levelValues && levelValues.length > 0)
+                                        content += this.buildSingleDimensionTooltip(level, levelValues);
+                                }
+                            }
+                        }
+                        else {
+                            content += this.buildSingleDimensionTooltip(dim, values);
+                        }
+                    }
+                }
+            }
+        }
+        content += '</div>';
+        return content;
+    },
+
+    buildSingleDimensionTooltip: function(dim, values) {
+        var colon = ': ', linebreak = '<br/>';
+        var value = this.buildTooltipValuesStr(values);
+        var label = Connector.getQueryService().getMeasure(dim).label;
+        return '<span><span class="axis-dimension">' + Ext.htmlEncode(label) + '</span>' + colon + value + linebreak + '</span>';
+    },
+
+    buildTooltipValuesStr: function(rawValuesArray) {
+        var valuesArray = [];
+        if (Ext.isArray(rawValuesArray)) {
+            if (rawValuesArray.length == 0)
+                return '-';
+            Ext.each(rawValuesArray, function(val) {
+                valuesArray.push(this.formatSingleTooltipValue(val));
+            }, this);
+        }
+        var valuesStr = '';
+        if (Ext.isArray(valuesArray)) {
+            var totalValues = valuesArray.length;
+            if (totalValues > 5) {
+                valuesStr = valuesArray.slice(0, 5).join(', ');
+                valuesStr += ', and ' + (totalValues - 5) + ' more'
+            }
+            else {
+                valuesStr = valuesArray.join(', ');
+            }
+        }
+        return valuesStr;
+    },
+
+    buildBinTooltip: function(plot, datas, plotData) {
```

Comment:
Minor changes made to buildBinTooltip to use queried plotData to generate Study tooltip instead of getting it from datas.